sql/pipelines/src/main/scala/org/apache/spark/sql/pipelines/logging/PipelineEvent.scala

```diff
@@ -0,0 +1,79 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.spark.sql.pipelines.logging
+
+import org.apache.spark.sql.pipelines.common.FlowStatus
+
+/**
+ * An internal event that is emitted during the run of a pipeline.
+ * @param id A time based, globally unique id
+ * @param timestamp The time of the event
+ * @param origin Where the event originated from
+ * @param message A user friendly description of the event
+ * @param details The details of the event
+ * @param error An error that occurred during the event
+ */
+case class PipelineEvent(
+    id: String,
+    timestamp: String,
+    origin: Origin,
+    message: String,
+    details: EventDetails,
+    error: Option[ErrorDetail]
+)
+
+/**
+ * Describes where the event originated from
+ * @param datasetName The name of the dataset
+ * @param flowName The name of the flow
+ * @param sourceCodeLocation The location of the source code
+ */
+case class Origin(
```

Comment:
Given there is an exiting Origin in Spark catalyst: https://github.com/apache/spark/blob/master/sql/api/src/main/scala/org/apache/spark/sql/catalyst/trees/origin.scala, shall we rename this one as `PipelineOrigin`?

Reply:
Updated to PipelineEventOrigin since the origin contains information not just about the pipeline but also about the dataset